planner/cascades/stringer.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cascades
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/pingcap/tidb/planner/memo"
+)
+
+// ToString stringifies a Group Tree.
+func ToString(g *memo.Group) []string {
```

Comment:
how about to implement the visitor pattern for `Group` and `GroupExpr`? Then we can use a visitor to collect the formatted explain informations.

Reply:
Should I implement visitor in another PR? I think this PR is big enough.

Reply:
ok